### PR TITLE
check for external program completion

### PR DESCRIPTION
**Pull request by bfjia**
partially solves issue 16 https://github.com/Public-Health-Bioinformatics/cpo-pipeline/issues/16

- check for existence of output from external programs before trying to parse the results
- moved all the file paths into a dictionary for organization

### CODE REVIEW

**Comment by bfjia** [comment]
2 new parsers is added for parsing file 'totalbp' and '$reference_genomeStats.txt' used for calculating coverage

File: assembly/pipeline.py [hunk 1]
```diff
@@ -52,17 +52,22 @@ def httpGetFile(url, filepath=""):
...
+        gzContent = f.read()
+    with open(file_path.replace(".gz",""), 'wb') as out:
+        out.write(gzContent)    
+    os.remove(file_path)
+    if (os.path.exists(file_path.replace(".gz",""))):
         return True
+    else:
+        raise Exception("Reference genome downloaded, but cannot be gunzipped")
+
+def check_file_exist_and_parse(path, parser):
```
Comment (dfornika) [inline assembly/pipeline.py:66]: What happens if we try to run a parser directly on a file that doesn't exist? Doesn't that throw an Exception already?

File: assembly/pipeline.py [hunk 2]
```diff
@@ -136,6 +141,18 @@ def main():
         "busco_complete_duplicate_cutoff":0.10 #BUSCO QC: complete duplicate genes less than ($thisvalue) percent will pass the QC
     }
     
+    file_paths = {
```
Comment (dfornika) [inline assembly/pipeline.py:144]: Could you give this a more descriptive name? Maybe `output_file_paths`?

File: assembly/pipeline.py [hunk 3]
```diff
@@ -144,12 +161,11 @@ def main():
     print("running pipeline_qc.sh")
     #input parameters: 1 = id, 2= forward, 3 = reverse, 4 = output, 5=mashgenomerefdb, $6=mashplasmidrefdb, $7=kraken2db, $8=kraken2plasmiddb
     cmd = [scriptDir + "/pipeline_qc.sh", ID, R1, R2, outputDir, mashdb, mashplasmiddb, kraken2db, kraken2plasmiddb]
-    result = execute(cmd, curDir)
+    #result = execute(cmd, curDir)
```
Comment (dfornika) [inline assembly/pipeline.py:164]: Why is this commented out?

**Comment by dfornika** [comment]
Thanks Justin @imasianxd !

**Pull request merged**